ComparisonWithOrleans.md

```diff
@@ -0,0 +1,203 @@
+# Orleans and Akka Actors: A Comparison
+
+## Step-By-Step Review
+
+This overview walks along the Technical Report [MSR-TR-2014-41](http://research.microsoft.com/apps/pubs/default.aspx?id=210931) by *Philip A. Bernstein, Sergey Bykov, Allan Geller, Cabriel Kliot, Jorgen Thelin (Microsoft Research)*, for another comparison focusing on Erlang and Riak see [Christopher Meiklejohn’s excellent article](http://christophermeiklejohn.com/orleans.html).
+
+### Introduction
+
+Different focus:
+
+  * Akka as a toolkit for building distributed systems, offering the full power but also exposing the inherent essential complexity.
+  
+  * Orleans restricts applicability in order to allow seamless use without understanding distributed computing.
+  
+#### Lifecycle
+
+  * Orleans Grains do not have a lifecycle, can neither be started nor stopped. Consequently they also cannot fail and be restarted, therefore Orleans does not offer tools for software failure handling—the failure handling aspects concentrate on recovering from hardware crashes.
```

Comment:
Orleans grain does have a lifecycle, it just doesn't allow external caller to control it. When the grain is activated we call OnActivateAsync, where the app code can do app specific initialization, such as read persistent storage. There is no way to explicitly pass arguments to OnActivateAsync, since the caller is not aware of the fact that grain has activation or not.
When we deactivate, we call OnDeactivateAsync, which can do a cleanup, flush persistent state, ...
Upon crash we indeed to do not invoke any application code (there is no "supervisor").